### ISSUE TITLE
Support simple key filter using spread operator

### ISSUE THREAD

**Issue by dimaqq** [eslint/eslint#13243]
**What rule do you want to change?**
`no-unused-vars`

**Does this change cause the rule to produce more or fewer warnings?**
fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
I don't know

**Please provide some example code that this change will affect:**

From: https://github.com/tc39/proposal-object-rest-spread/issues/16#issuecomment-233218801

To filter out the key `md5` from `x` with the result in `y`, do:

```js
const x = {a: 1, b: 2, c: 3, md5: '83b0f2d70d7546ee634f801264c4a9b3'};
const {md5, ...y} = x;
```

**What does the rule currently do for this code?**
`no-unused-vars: 'md5' is assigned a value but never used.`

**What will the rule do after it's changed?**
nothing

**Are you willing to submit a pull request to implement this change?**
First, let's decide if it's a good idea :)

Perhaps `const {m5: bit_bucket, ...y} = ...` would be better, if there was a bit bucket?

**Comment by kaicataldo** [comment]
Just to confirm, are you taking about the rest operator? If so, this functionality is already available using the [ignoreRestSiblings](https://eslint.org/docs/rules/no-unused-vars#ignorerestsiblings) configuration option.

**Comment by dimaqq** [comment]
Ah, you are right, @kaicataldo thank you!

**Comment by kaicataldo** [comment]
No problem! Glad you were able to resolve it 😄

**Comment by kaicataldo** [comment]
Closing this issue since the question has been answered.
